pyfr/util.py

```diff
@@ -96,6 +96,48 @@ def __exit__(self, *args):
         os.close(self.saved_fds[1])
 
 
+class Interval:
```

Comment:
I'm still not sure this class is needed.  Merge intervals will cause duplicate time regions.

Reply:
It is used in `_active_files` to reduce the list of files to be read to just those needed. For example continuous mode time averaging followed by one windowed mode at a later time could give files with the intervals [(0,1), (0,2), (3,4)].
merge_intervals will fail.

Reply:
That should probably be an error. Silently trying to correct our users mistakes typically never ends well.  We should error, and then let the user retry the command without the duplicate file range (or realise that they've done something else wrong and rethink their course of action).